File: client-app/src/app/models/comment.ts
export interface Comment {
  id: string;
  content: string;
  createdAt: Date;
  updatedAt?: Date;
  ticketId: string;
  authorId: string;
  authorUsername: string;
  authorDisplayName: string;
  attachments: CommentAttachment[];
}

export interface CommentAttachment {
  id: string;
  fileName: string;
  originalFileName: string;
  contentType: string;
  fileSize: number;
  uploadedAt: Date;
  downloadUrl: string;
}

export interface CreateCommentForm {
  content: string;
  ticketId: string;
  attachments: File[];
}

File: Domain/Comment.cs
using System;

namespace Domain
{
    public class Comment
    {
        public Guid Id { get; set; }
        public string Content { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public Guid TicketId { get; set; }
        public Ticket Ticket { get; set; }
        public string AuthorId { get; set; }
        public AppUser Author { get; set; }
        public Guid? ParentCommentId { get; set; }
        public Comment ParentComment { get; set; }
        public ICollection<Comment> Replies { get; set; } = new List<Comment>();
        public ICollection<CommentAttachment> Attachments { get; set; } = new List<CommentAttachment>();
    }
}